easybuild/framework/easyblock.py

```diff
@@ -2037,10 +2037,16 @@ def install_all_extensions(self, install=True):
         self.log.debug("List of loaded modules: %s", self.modules_tool.list())
 
         if build_option('parallel_extensions_install'):
-            try:
+            # check to see if parallel extension install is supported for all extensions
+            # if it is not then we'll fallback to sequential install
+            all_exts_parallel = True
+            for ext in self.ext_instances:
+                if Extension.install_extension_async.__code__.co_code == ext.install_extension_async.__code__.co_code:
+                    all_exts_parallel = False
```

Comment:
As soon as one extension can't be installed in parallel, we don't need to check any further:

```suggestion
                    all_exts_parallel = False
                    break
```